src/cli/deadend.rs

```diff
@@ -28,6 +29,7 @@ pub enum Cmd {
 impl Cmd {
     /// `deadend-motd` subcommand entry point.
     pub(crate) fn run(self) -> Fallible<()> {
+        ensure_zincati_user("deadend-motd subcommand must be run by `zincati` user")?;
```

Comment:
Uhm, this doesn't look correct though. This helper is run through `pkexec`, so I would expect this to always run as `root`. Did you check this either manually or via kola tests?

Reply:
ah I see, my bad. I only checked that the command couldn't be run as my normal user. Hmm I think the invocation ID environment variable won't work here either because `pkexec` doesn't pass on environment variables in the `exec`'ed program.

Reply:
Could we get `pkexec` to execute as the `zincati` user then?

Reply:
No, wait, those are separate concerns:
 * `zincati agent` runs as `zincati` user under a systemd service unit
 * `zincati deadend-motd` runs as `root` user under pkexec

Their respective setup is already ok, it's only this specific check which is wrong. The original ticket was correctly scoped to the `agent` verb only, but the PR dragged the check into an additional place where it doesn't belong.

Reply:
The problem with this PR (the `deadend-motd` check) is that it'll fail even when it was the `zincati agent` (running as the `zincati` user) that `pkexec`'ed `/usr/libexec/zincati deadend-motd` right? The check will cause `/usr/libexec/zincati deadend-motd` to exit because it sees that it is running as `root`. So I thought that if we made `pkexec` run as `zincati` user, then the happy path would work; and we'll still be able to exit early if a user tries to run `/usr/libexec/zincati deadend-motd`.

Reply:
Or we could just skip the check for `zincati deadend-motd` and if a user tries to run that subcommand, then s/he'll get 
```
$ /usr/libexec/zincati deadend-motd set --reason foo
[ERROR] critical error: failed to create temporary MOTD file under '/run/motd.d/'
[ERROR]  -> Permission denied (os error 13) at path "/run/motd.d/.deadend.acAhvM.motd.partial"
```
(I think this is what you mean right?)

Reply:
Yes, we can either drop the check at all (previous status quo), or fix it to match the real expected user.

Reply:
Sorry about this BTW!  I really went back and forth with myself for a minute before clicking the merge button.  My thought process flowed like: "I don't know zincati really well, but this patch looks sane.  And I think it's covered by CI tests.  Should it wait for luca?  It could, but on the other hand it's good to build up a sense of shared ownership in projects and not have one person every PR waits for.  Hmm, well if in the end something isn't right he should see this and be able to review post-merge."

Which is basically what happened :smile:  Curious what you all think about this.

Reply:
To be fair, I really should've understood what `pkexec` does first... apologies for that. This PR also unfortunately wasn't covered by CI (working on that now).
I think having non-Luca people review (and approve) smaller PRs like this one makes sense, so we don't spam him with every little thing. 
But I'll defer to Luca on whether we should defer to Luca for everything.

Reply:
No worries, I think the flow here was fine, and as Kelvin said we mostly hit a CI gap.

Overall kola does cover the critical path, so in the worst case a green merge should at most regress some minor/UX feature, but the core upgrading logic should be fine.